docs/madr/decisions/039-transition-to-new-policies.md

```diff
@@ -0,0 +1,644 @@
+
+* `GET /meshes/{mesh}/dataplanes/{name}/_snapshot` -> `GET /meshes/{mesh}/dataplanes/{name}/_snapshot?shadow=true`
+* `GET /meshes/{mesh}/dataplanes/{name}/_rules` -> `GET /meshes/{mesh}/dataplanes/{name}/_rules?shadow=true`
+
+when `shadow=true` is set, the response will take policies with `kuma.io/effect: shadow` label into account.
+
+For the debugging purposes it makes sense to represent the shadow snapshot as the original snapshot + jsonPatch that can be applied to it.
+Kuma GUI and kumactl also can benefit from such representation. To make server produce such response we can introduce a query parameter `include=diff`.
+Query param `include` contains a list of fields that should be included in the response. By default, only `xds` field is set by the API server. 
+From the validation point of view the request `_snapshot?shadow=true&include=diff` makes no sense and should be rejected.  
+
+OpenAPI spec for new endpoints:
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Shadow Policies Inspector
+  description: API for inspecting policies' effect on the Data Plane Proxy configurations
+  version: 1.0.0
+paths:
+  /meshes/{mesh}/dataplanes/{name}/_snapshot:
+    get:
+      summary: Get a DPP snapshot on a CP
+      description: Returns a DPP snapshot computed by the CP without the actual Envoy call.
+      parameters:
+        - in: path
+          name: mesh
+          required: true
+          description: The mesh of the DPP to get the diff for.
+          schema:
+            type: string
+        - in: path
+          name: name
+          required: true
+          description: The name of the DPP within the mesh to get the diff for.
+          schema:
+            type: string
+        - in: query
+          name: shadow
+          description: |
+            When computing XDS config the CP takes into account policies with 'kuma.io/effect: shadow' label
+          schema:
+            type: boolean
+            default: false
+        - in: query
+          name: include
+          description: |
+            An array of field names that also should be included in the response. By default, only 'xds' field is set by the API server. 
+          schema:
+            type: array
+            items:
+              type: string
+              enum: [diff]
+      responses:
+        '200':
+          description: Successfully retrieved snapshot.
+          content:
+            application/json:
+              schema:
+                type: object
+                required: [xds]
+                properties:
+                  xds:
+                    description: The raw XDS config as an inline JSON object
+                    type: object
+                  diff:
+                    description: |
+                      When '?include=diff' is specified, the field is set to an array of json patches between the current XDS config
+                      and XDS config that takes into account policies with 'kuma.io/effect: shadow'.
```

Comment:
This doesn't actually say what the output is. This is a list of json patches so we should likely link to the spec of json patch for example.

Put yourself in the user's shoes, what do you think they'd want to know?

Reply:
But we have a json patch spec right here https://github.com/kumahq/kuma/blob/7d5702d38fae50ac72021d0225dbd77593a45274/docs/madr/decisions/039-transition-to-new-policies.md?plain=1#L130-L146

I can add a link to the RFC https://datatracker.ietf.org/doc/html/rfc6902/

Reply:
Only thing that I was saying is that your doc doesn't say it's a JSONPatch sure the spec is here but folks have to parse/understand it whereas if they just see "it's a JSONPatch" they get it quicker.